applications/FluidDynamicsApplication/tests/test_mass_conservation_check_process.py

```diff
@@ -0,0 +1,223 @@
+        self.model = KratosMultiphysics.Model()
+        model_part = self.model.CreateModelPart("ModelPart")
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.MESH_VELOCITY)
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.VELOCITY)
+        model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISTANCE)
+        if self.comm.IsDistributed():
+            model_part.AddNodalSolutionStepVariable(KratosMultiphysics.PARTITION_INDEX)
+
+        model_part.SetBufferSize(3)
+
+        with KratosUnittest.WorkFolderScope(self.work_folder, __file__):
+            import_settings = KratosMultiphysics.Parameters(self.parameters)
+            self._ImportModelPart(model_part, import_settings)
+
+        return model_part
+
+    def _ImportModelPart(self, model_part, import_settings):
+        if self.comm.IsDistributed():
+            distributed_model_part_importer = distributed_import_model_part_utility.DistributedImportModelPartUtility(model_part, import_settings)
+            distributed_model_part_importer.ImportModelPart()
+            distributed_model_part_importer.CreateCommunicators()
+        else:
+            import_flags = KratosMultiphysics.ModelPartIO.READ
+            KratosMultiphysics.ModelPartIO(import_settings['model_import_settings']['input_filename'].GetString(), import_flags).ReadModelPart(model_part)
+
+
+    def test_Initialize(self):
+        self.work_folder = "auxiliary_files"
+        self.file_name = "cube_few_elements"
+        model_part = self._CreateModelPart()
+        for node in model_part.GetCommunicator().LocalMesh().Nodes:
+            node.SetSolutionStepValue(KratosMultiphysics.DISTANCE, node.Y - 0.5)
+        settings = KratosMultiphysics.Parameters()
+        mass_conservation_process = KratosFluid.MassConservationUtility(model_part, settings)
+        message = mass_conservation_process.Initialize()
+        filtered_results = re.findall(r"[-+]?\d*\.\d+|\d+", message)
+        self.assertAlmostEqual(float(filtered_results[0]), 0.5)
+        self.assertAlmostEqual(float(filtered_results[1]), 0.5)
+        self.assertAlmostEqual(float(filtered_results[2]), 1.0)
+
+    def test_ComputeBalancedVolume(self):
+        self.work_folder = "auxiliary_files"
+        self.file_name = "cube_few_elements"
+        model_part = self._CreateModelPart()
+        for node in model_part.GetCommunicator().LocalMesh().Nodes:
+            node.SetSolutionStepValue(KratosMultiphysics.DISTANCE, node.Y - 0.5)
+            node.SetSolutionStepValue(KratosMultiphysics.VELOCITY_X, 1.0 - 0.5*node.X)
+        inlet_conds = [5,6,17,18,29,30,41,42]
+        outlet_conds = [11,12,23,24,35,36,47,48]
+        for cond in model_part.Conditions:
+            if cond.Id in inlet_conds:
+                cond.Set(KratosMultiphysics.INLET)
+            if cond.Id in outlet_conds:
+                cond.Set(KratosMultiphysics.OUTLET)
```

Comment:
If I'm not wrong, this is copied and pasted below. I suggest creating an auxiliary function.

Reply:
you are right, corrected